metadata.go

```diff
@@ -0,0 +1,23 @@
+package iavl
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// Marshal returns Protobuf marshaled bytes of a VersionMetadata object.
+func (vm *VersionMetadata) Marshal() ([]byte, error) {
+	return proto.Marshal(vm)
+}
+
+// Unmarshal attempts to decode a VersionMetadata object from Protobuf marshaled
+// bytes.
+func (vm *VersionMetadata) Unmarshal(bz []byte) error {
+	return proto.Unmarshal(bz, vm)
+}
+
+// VersionMetadataKey returns the indexing key for a VersionMetadata object.
+func VersionMetadataKey(version int64) []byte {
+	return []byte(fmt.Sprintf("metadata/%d", version))
```

Comment:
We should use a binary encoding which retains the natural ordering of versions and makes it possible to scan across them. `key_format.go` has tooling for this, I'd suggest prefix `m` or `v`.

Reply:
I didn't imagine we'd have a need to range of these -- I'm still not sure why we would, we don't even expose an API for this. But we can leverage the existing key formatting logic.